reconcilers/reconcilers_test.go

```diff
@@ -605,6 +858,69 @@ func TestChildReconciler(t *testing.T) {
 		ExpectDeletes: []rtesting.DeleteRef{
 			{Group: "", Kind: "ConfigMap", Namespace: testNamespace, Name: testName},
 		},
+	}, {
+		Name: "delete child, clearing finalizer",
+		Parent: resourceReady.
+			MetadataDie(func(d *diemetav1.ObjectMetaDie) {
+				d.Finalizers(testFinalizer)
+			}),
+		GivenObjects: []client.Object{
+			configMapGiven,
+		},
+		Metadata: map[string]interface{}{
+			"SubReconciler": func(t *testing.T, c reconcilers.Config) reconcilers.SubReconciler {
+				r := defaultChildReconciler(c)
+				r.Finalizer = testFinalizer
+				return r
+			},
+		},
```

Comment:
*[question]:* The doc for `Metadata` is _"Metadata contains arbitrary value that are stored with the test case"_. Why attach a function here?